File: Frontend/E-commerce.web/src/app/services/cart.service.ts
import { HttpClient } from "@angular/common/http";
import { inject, Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { Product } from "./product.service";
export interface CartItem {
  id: number;
  productId: number;
  productName: string;
  price: number;
  quantity: number;
  imageUrl: string;
}
@Injectable({
  providedIn: 'root'
})
export class CartService {
  private apiUrl = 'http://localhost:5000/api/cart';
  private http = inject(HttpClient);
  


  getCartItems(): Observable<CartItem[]> {
    return this.http.get<CartItem[]>(this.apiUrl);
  }

  addToCart(item: CartItem): Observable<CartItem> {
    return this.http.post<CartItem>(this.apiUrl, item);
  }

  updateQuantity(productId: number, quantity: number): Observable<CartItem> {
    return this.http.put<CartItem>(`${this.apiUrl}/${productId}`, quantity);
  }

  removeFromCart(productId: number): Observable<CartItem> {
    return this.http.delete<CartItem>(`${this.apiUrl}/${productId}`);
  }

  clearCart(): Observable<CartItem> {
    return this.http.delete<CartItem>(this.apiUrl);
  }
}

File: E-commerce/Services/CartService.cs
using E_commerce.Models.Domain;

namespace E_commerce.Services
{
    public class CartService
    {
        private readonly List<CartItem> _cartItems = new();
        public List<CartItem> GetCartItems() => _cartItems;

        public void AddToCart(CartItem item)
        {
            var existingItem = _cartItems.FirstOrDefault(ci => ci.ProductId == item.ProductId);
            if (existingItem != null)
            {
                existingItem.Quantity += item.Quantity;
            }
            else
            {
                _cartItems.Add(item);
            }
        }
        public void UpdateCartItem(int productId, int quantity)
        {
            var item = _cartItems.FirstOrDefault(ci => ci.ProductId == productId);
            if (item != null)
            {
                item.Quantity = quantity;
            }
        }
        public void RemoveFromCart(int productId)
        {
            _cartItems.RemoveAll(ci => ci.ProductId == productId);
        }

        public void ClearCart()
        {
            _cartItems.Clear();
        }

    }
}
